tests/test_upload.py

```diff
@@ -58,6 +59,82 @@ def test_successful_upload(make_settings, capsys):
     assert captured.out.count(NEW_RELEASE_URL) == 1
 
 
+def test_successful_upload_add_gpg_signature(make_settings, monkeypatch):
+    """Test uploading package when gpg signature are added to it"""
+
+    # Patch create_repository() to mock successful upload of package
+    # and allow us to make assertions on the uploaded package
+
+    stub_response = pretend.stub(
+        is_redirect=False, status_code=201, raise_for_status=lambda: None
+    )
+
+    stub_repository = pretend.stub(
+        upload=pretend.call_recorder(lambda package: stub_response),
+        close=lambda: None,
+        release_urls=lambda _: None,
+    )
+
+    upload_settings = make_settings()
+    upload_settings.create_repository = lambda: stub_repository
+
+    # Upload a pre-signed distribution
+    result = upload.upload(
+        upload_settings, [helpers.WHEEL_FIXTURE, helpers.WHEEL_FIXTURE + ".asc"]
+    )
+    assert result is None
+
+    # The signature shoud be added via package.add_gpg_signature()
+    package = stub_repository.upload.calls[0].args[0]
+    assert package.gpg_signature == (
+        "twine-1.5.0-py2.py3-none-any.whl.asc",
+        b"signature",
+    )
+
+
+def test_successful_upload_sign_package(make_settings, monkeypatch):
+    """Test uploading package when package is signed"""
+
+    # Patch create_repository() to mock successful upload of package
+    # and allow us to make assertions on the uploaded package
+
+    stub_response = pretend.stub(
+        is_redirect=False, status_code=201, raise_for_status=lambda: None
+    )
+
+    stub_repository = pretend.stub(
+        upload=pretend.call_recorder(lambda package: stub_response),
+        close=lambda: None,
+        release_urls=lambda _: None,
+    )
+
+    upload_settings = make_settings()
+    upload_settings.create_repository = lambda: stub_repository
```

Comment:
So we are going to extract this part into a fixture which will live within `test_upload` and will make this more readable, or is this something which can be moved to `conftest.py` is this pattern is repeated in other tests as well?